source/usage-examples/watch.txt

```diff
@@ -0,0 +1,139 @@
+=================
+Watch for Changes
+=================
+
+.. default-domain:: mongodb
+
+Open a Change Stream
+--------------------
+
+You can keep track of changes to data in MongoDB, such as changes to a
+collection, database, or deployment, by opening a **change stream**. A change
+stream allows applications to watch for changes to data and react to them.
+You can open a change stream by calling the ``watch()`` method on
+a ``MongoCollection``, ``MongoDatabase``, or ``MongoClient`` object. The
+change stream returns **change event** documents when they occur.
+
+The ``watch()`` method optionally takes an **aggregation pipeline**  which
+consists of an array of **stages** as the first parameter to filter and
+transform the change event output as follows:. 
+
+.. code-block:: java
+
+   List<Bson> pipeline = Arrays.asList(
+                           Aggregates.match(
+                              Filters.lt("fullDocument.runtime", 15)));
+   ChangeStreamIterable<Document> changeStream = database.watch(pipeline);
+
+To configure options for the change stream, use member methods of the
+``ChangeStreamIterable`` object returned by ``watch()``. Refer to the
+links at the end of this section for more information on the settings
+you can configure.
+
+The ``watch()`` method returns an instance of a ``ChangeStreamIterable``.
+You can call ``forEach()`` on the ``ChangeStreamIterable`` to handle
+events as they occur, or you can use the ``iterator()`` method to access
+the underlying ``MongoIterator``. This allows you to call manual methods
+on the change stream such as ``hasNext()`` to check for remaining
+documents in the stream and ``next()`` to request the next document in
+the stream. See the link to the ``ChangeStreamIterable`` API
+documentation below for more details on the available methods.
+
+Visit the following resources for additional material on the classes and
+methods presented above:
+
+- :manual:`change streams </changeStreams/>`
+- :manual:`change events </reference/change-events/>`
+- :manual:`aggregation pipeline </reference/operator/aggregation-pipeline/>`
+- :manual:`aggregation stages </changeStreams/#modify-change-stream-output>`
+- :java-sync-api:`ChangeStreamIterable <com/mongodb/client/ChangeStreamIterable.html>`
+- :java-sync-api:`MongoCollection.watch() <com/mongodb/client/MongoCollection.html#watch()>`
+- :java-sync-api:`MongoDatabase.watch() <com/mongodb/client/MongoDatabase.html#watch()>`
+- :java-sync-api:`MongoClient.watch() <com/mongodb/client/MongoClient.html#watch()>`
+
+Process the Change Stream Events
```

Comment:
Nitpick: Similar comment to the one under the "Open a Change Stream" heading. I like the material itself, but perhaps it could be labeled as "How to Process Change Stream Events with a Callback", or just be rolled up into the Example. Alternatively, we could move it to Fundamentals content on change streams. It might not be the right material for Jasmin, but definitely appropriate for Lucas.

Reply:
Considered removing both headings, but I think this section is too chunky without something to break it up. Taking your suggestion.